application/models/SurveysGroupsettings.php

```diff
@@ -292,21 +301,24 @@ public function getIsAnonymized()
      * It steps up (see param $iStep) until it has found the real settings ...
      *
      * @param int $iSurveyGroupId
-     * @param null $oSurvey
-     * @param null $instance
+     * @param  \Survey|null $oSurvey
+     * @param \self|null $instance
      * @param int $iStep      this is inheritance step (recursive step) (parent, parentParent, parentParentParent ?)
      * @param bool $bRealValues
      * @return SurveysGroupsettings instance
      */
     public static function getInstance($iSurveyGroupId = 0, $oSurvey = null, $instance = null, $iStep = 1, $bRealValues = false)
     {
 
-        if ($iSurveyGroupId > 0) {
-            $model = SurveysGroupsettings::model()->with('SurveysGroups')->findByPk($iSurveyGroupId);
-        } else {
-            //this is the default group setting with gsid=0 !!!
-            $model = SurveysGroupsettings::model()->findByPk($iSurveyGroupId);
+        if (!array_key_exists($iSurveyGroupId, self::$aSurveysGroupSettings)) {
```

Comment:
issset() wou;dn't be faster than array_key_exists?

Reply:
> issset() wou;dn't be faster than array_key_exists?

It can be null here : https://github.com/LimeSurvey/LimeSurvey/blob/c16d1df55a68853c3c9c4f15d9640d6799b050b0/application/models/SurveysGroupsettings.php#L325

Yes isset is better if data is not null :)

Reply:
Right.